src/linker/Linker/Pipeline.cs

```diff
@@ -140,6 +150,11 @@ public IStep[] GetSteps ()
 			return _steps.ToArray ();
 		}
 
+		public IAssemblyStep[] GetAssemblySteps ()
```

Comment:
It's not great to allocate array on every call for data which are static. You could just have the loop here.